src/main/java/ru/itmo/sd/deadliner2bot/repository/DailyNotificationRepository.java

```diff
@@ -10,14 +9,17 @@
 
 public interface DailyNotificationRepository extends CrudRepository<DailyNotification, Long> {
 
-    @Query(value = "SELECT * FROM daily_notifications WHERE notification_time < ?", nativeQuery = true)
-    Set<DailyNotification> findAllBeforeTimeLimit(LocalDateTime timeLimit);
-
-    @Modifying
-    @Query(value = "DELETE FROM daily_notifications WHERE notification_time < ?", nativeQuery = true)
-    void removeAllBeforeTimeLimit(LocalDateTime timeLimit);
+    @Query(value = "SELECT * FROM daily_notifications " +
+            "WHERE EXTRACT(HOUR FROM notification_time) = EXTRACT(HOUR FROM CAST(?1 AS TIMESTAMP))" +
```

Comment:
Seems to me, that this is not ideal check. 
For example, if `now` time is `11:59:30` (we use `now` with seconds precision), limit is `12:00:30`. If we have notification with `12:00` (in database time with minute precision), it won't fetch. Situation will be worse if we want to increase fetching limit in future.
If it is true, my way to fix it is to calculate diff between timestamps and take it modulo week (7 * 24 * 60 * 60 sec)

Reply:
Fetching will happen at `xx:xx:00` + eps, we changed the signature of the scheduled task.

Reply:
And how it helps? change 30 to eps, and we still have 12:00:eps limit. And still problem with increasing fetching limit.

Reply:
How do you suggest putting a configurable variable into the annotation valur?

Reply:
Same question regarding the scheduled task. We currently have no means to externally configure fetching delay.

Reply:
I do not understand you. My question wasn't about fetching delay as configurable variable. My main point is that checking logic is incorrect as I wrote above. Do you agree?

Reply:
No, it works fine in the current configuration. `now` will never be anything but something like `xx:xx:00` + eps because we changed the configuration of `@Scheduled` method, now it uses `cron` parameter. Thus, it will always fetch everything in this specific minute. 
You've still got a point that we should use a more robust check.